packages/react-widgets/src/models/TableModel.js

```diff
@@ -6,14 +6,15 @@ export function getTable(props) {
 }
 
 function fromLocal(props) {
-  const { source, rowsPerPage, page, sortBy, sortDirection } = props;
+  const { source, rowsPerPage, page, sortBy, sortDirection, sortByColumnType } = props;
```

Comment:
```suggestion
  // Injecting sortByColumnType externally from metadata gives better results. It allows to avoid deriving type from row value itself (with potential null values)
  const { source, rowsPerPage, page, sortBy, sortDirection, sortByColumnType } = props;
```